src/contracts/interfaces/IStakeRegistry.sol

```diff
@@ -0,0 +1,148 @@
+        // the block number at which the stake amounts were updated and stored
+        uint32 updateBlockNumber;
+        // the block number at which the *next update* occurred.
+        /// @notice This entry has the value **0** until another update takes place.
+        uint32 nextUpdateBlockNumber;
+        // stake weight for the quorum
+        uint96 stake;
+    }
+
+    /**
+     * @notice Registers the `operator` with `operatorId` for the specified `quorumNumbers`.
+     * @param operator The address of the operator to register.
+     * @param operatorId The id of the operator to register.
+     * @param quorumNumbers The quorum numbers the operator is registering for.
+     * @dev Permissioned by RegistryCoordinator
+     */
+    function registerOperator(address operator, bytes32 operatorId, uint8[] memory quorumNumbers) external;
+
+    /**
+     * @notice Deregisters the operator with `operatorId` for the specified `quorumNumbers`.
+     * @param operatorId The id of the operator to deregister.
+     * @param quorumNumbers The quourm numbers the operator is deregistering from.
+     * @dev Permissioned by RegistryCoordinator
+     */
+    function deregisterOperator(bytes32 operatorId, uint8[] memory quorumNumbers) external;
+
+    function getLengthOfTotalStakeHistoryForQuorum(uint8 quorumNumber) external view returns (uint256);
+
+    /**
+     * @notice Returns the `index`-th entry in the dynamic array of total stake, `totalStakeHistory` for quorum `quorumNumber`.
+     * @dev Function will revert in the event that `index` is out-of-bounds.
+     */
+    function getTotalStakeUpdateForQuorumFromIndex(uint8 quorumNumber, uint256 index) external view returns (OperatorStakeUpdate memory);
+
+    /**
+     * @notice Returns the `index`-th entry in the `operatorIdToStakeHistory[operatorId][quorumNumber]` array.
+     * @param quorumNumber The quorum number to get the stake for.
+     * @param operatorId The id of the operator of interest.
+     * @param index Array index for lookup, within the dynamic array `operatorIdToStakeHistory[operatorId][quorumNumber]`.
+     * @dev Function will revert if `index` is out-of-bounds.
+     */
+    function getStakeUpdateForQuorumFromOperatorIdAndIndex(uint8 quorumNumber, bytes32 operatorId, uint256 index)
+        external
+        view
+        returns (OperatorStakeUpdate memory);
+
+    /**
+     * @notice Returns the stake weight corresponding to `operatorId` for quorum `quorumNumber`, at the
+     * `index`-th entry in the `operatorIdToStakeHistory[operatorId][quorumNumber]` array if it was the operator's
+     * stake at `blockNumber`. Reverts otherwise.
+     * @param quorumNumber The quorum number to get the stake for.
+     * @param operatorId The id of the operator of interest.
+     * @param index Array index for lookup, within the dynamic array `operatorIdToStakeHistory[operatorId][quorumNumber]`.
+     * @param blockNumber Block number to make sure the stake is from.
```

Comment:
perhaps this is a good place to clarify better on the exact expected usage of this input

Reply:
same issue is present in `getTotalStakeAtBlockNumberFromIndex`

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/48/commits/7a220382b4fdbe6880cbec7103c3e5db6a036f24